### DISCUSSION TITLE
Feature request: some more debugging insights

### DISCUSSION THREAD

**Discussion by robdewit** [opentofu/opentofu#1779] [Ideas]
As much as working with Tofu can be a great experience, it sometimes can also be quite frustrating when you get some "computer says no" response.

For example for some complex `for_each` construct using a combination of *setproduct*, *tomap*, etc. the response while doing a *plan* or even only when you do *apply* gives an error about the value not being map or set, but I have found no way of actually checking what it is that actually got produced by the construct. So I'm left with two options:

1) try to guess which part is wrong
2) meticulously replaying whatever it was that I wrote, which can be challenging it the objects themselves have been produced by other pieces of code

I would find it very helpful if there was a debugging level that just dumped the object structure that is being evaluated. This could be some pragma in code to specifically dump troublesome code, or a global setting dumping everything that is being evaluated in the run.

Maybe this option already exists, but I have not been able to find it.

**Comment by ghost** [comment]
Hi @robdewit could you open an issue for the object structure dumping please? We implement features based on issues.
